src/event_object_movement.c

```diff
@@ -4008,6 +4035,17 @@ void SetObjectEventDirection(struct ObjectEvent *objectEvent, u8 direction)
 
 static const u8 *GetObjectEventScriptPointerByLocalIdAndMap(u8 localId, u8 mapNum, u8 mapGroup)
 {
+    switch (localId)
+    {
+        case 0xF0:
+            return gSaveBlock1Ptr->dynamicObjects[0].scriptPtr;
+        case 0xF1:
+            return gSaveBlock1Ptr->dynamicObjects[1].scriptPtr;
+        case 0xF2:
+            return gSaveBlock1Ptr->dynamicObjects[2].scriptPtr;
+        case 0xF3:
+            return gSaveBlock1Ptr->dynamicObjects[3].scriptPtr;
```

Comment:
Possible to do localid - 0xF0 * here, so it's easier to change the dynamic npc array size constant